src/operators/delay.ts

```diff
@@ -79,7 +79,7 @@ interface DelayState<T> {
  */
 class DelaySubscriber<T> extends Subscriber<T> {
   private queue: Array<DelayMessage<T>> = [];
-  private active: boolean = false;
+  private nextAction: Subscription = null;
```

Comment:
i'm suspicious this will have issues in situations where more than one delayed action is dispatched synchronously. For example `of(1, 2, 3).pipe(delay(10))` or the like. You'd have to keep more than one subscription for each scheduled action.

Reply:
The original code would only schedule one action. `this.active` would be set to `true` and additional messages would be buffered until dispatch. On dispatch the queue would be processed... etc. I hope I am interpreting it correctly...

I think this code does the same except it doesn't register unsubscription logic that is the cause of the leak.